src/contexts/GustoApiProvider/useGustoProvider.tsx

```diff
@@ -24,31 +18,32 @@ export type Dictionary = Record<
   Partial<{ [K in keyof Resources]: DeepPartial<Resources[K]> }>
 >
 
-export interface GustoApiProps {
+export interface GustoProviderProps {
   config: APIConfig
   dictionary?: Dictionary
   lng?: string
   locale?: string
   currency?: string
   theme?: DeepPartial<GTheme>
-  children?: React.ReactNode
   queryClient?: QueryClient
+  components?: Partial<ComponentsContextType>
 }
 
-const GustoApiProvider: React.FC<GustoApiProps> = ({
+export const useGustoProvider = ({
```

Comment:
Nit: wondering if it's worth the indirection to move this to the hook vs. just keeping this in GustoProviderCustomUIAdapter? We only end up with one usage of this now that GustoProvider just brings in the GustoProviderCustomUIAdapter